packages/@aws-cdk/aws-apigatewayv2-integrations/test/http/aws.test.ts

```diff
@@ -0,0 +1,86 @@
+import '@aws-cdk/assert-internal/jest';
+import { HttpApi, HttpRoute, HttpRouteKey } from '@aws-cdk/aws-apigatewayv2';
+import { StateMachine, Chain, Pass, StateMachineType } from '@aws-cdk/aws-stepfunctions';
+import { Stack, Duration } from '@aws-cdk/core';
+import { StepFunctionsStartExecutionIntegration, StepFunctionsStartSyncExecutionIntegration } from '../../lib';
+
+describe('AwsServiceIntegration', () => {
+  test('StepFunctions-StartExecution', () => {
+    const stack = new Stack();
+    const api = new HttpApi(stack, 'HttpApi');
+
+    new HttpRoute(stack, 'StepFunctionsStartExeRoute', {
+      httpApi: api,
+      integration: new StepFunctionsStartExecutionIntegration(stack, {
+        stateMachine: stateMachine(stack),
+        name: 'MyExe',
+        input: '$request.body',
+        timeout: Duration.seconds(10),
+        description: 'Start execution of state machine',
+      }),
+      routeKey: HttpRouteKey.with('/start'),
+    });
+
+    expect(stack).toHaveResource('AWS::ApiGatewayV2::Integration', {
+      IntegrationType: 'AWS_PROXY',
+      CredentialsArn: {
+        'Fn::GetAtt': [
+          'StepFunctionsStartExeRouteApiGatewayIntegrationRoleE88875A9',
+          'Arn',
+        ],
+      },
+      IntegrationSubtype: 'StepFunctions-StartExecution',
+      PayloadFormatVersion: '1.0',
+      RequestParameters: {
+        StateMachineArn: {
+          Ref: 'MyStateMachine6C968CA5',
+        },
+        Input: '$request.body',
+        Name: 'MyExe',
+      },
+      TimeoutInMillis: 10000,
+    });
+  });
+
+  test('StepFunctions-StartSyncExecution', () => {
+    const stack = new Stack();
+    const api = new HttpApi(stack, 'HttpApi');
+
+    new HttpRoute(stack, 'StepFunctionsStartSyncExeRoute', {
+      httpApi: api,
+      integration: new StepFunctionsStartSyncExecutionIntegration(stack, {
+        stateMachine: new StateMachine(stack, 'MyStateMachine', {
+          stateMachineName: 'MyStateMachine',
+          definition: Chain.start(new Pass(stack, 'Pass')),
+          stateMachineType: StateMachineType.EXPRESS,
+        }),
+        input: {
+          a: 'b',
+        },
+      }),
+      routeKey: HttpRouteKey.with('/startSync'),
+    });
+
+    expect(stack).toHaveResource('AWS::ApiGatewayV2::Integration', {
+      IntegrationType: 'AWS_PROXY',
+      IntegrationSubtype: 'StepFunctions-StartSyncExecution',
+      PayloadFormatVersion: '1.0',
+      RequestParameters: {
+        StateMachineArn: {
+          Ref: 'MyStateMachine6C968CA5',
+        },
+        Input: {
+          a: 'b',
+        },
+      },
+    });
```

Comment:
should you validate if the other resources are also created? The IAM Role for example

Reply:
IAM Role did not show up in the properties when testing, but I will add more extended tests for both.